tests/integration/disambiguation/test_disambiguation_api.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+from mock import patch
+from tempfile import mkstemp
+
+from flask import current_app
+
+from factories.db.invenio_records import TestRecordMetadata
+
+from inspirehep.modules.disambiguation.api import save_training_data
+
+
+def test_save_training_data(isolated_app):
+    TestRecordMetadata.create_from_file(__name__, '792017.json')
+
+    clusters_temporary_fd, clusters_path = mkstemp()
```

Comment:
why not use [`tmpdir`](https://docs.pytest.org/en/latest/tmpdir.html)? It's used e.g. [here](https://docs.pytest.org/en/latest/tmpdir.html) and handles all the mess for you.

Reply:
Excellent idea. Using it simplified the test a great deal.

Reply:
I liked it so much that I added https://github.com/inspirehep/inspire-next/pull/3422/commits/4df1c3bba93b102e299b2617745b956a64fc218c to rewrite all tests that were using `mkdtemp` / `mkstemp` so that we only have correct examples in the code base!